tesseract-server/src/handlers/logic_layer/aggregate.rs

```diff
@@ -71,124 +133,188 @@ pub fn logic_layer_aggregation(
+        info!("Tesseract query: {:?}", ts_query);
 
-            for time_cut in time_cuts {
-                let tc: Vec<String> = time_cut.split(".").map(|s| s.to_string()).collect();
+        let query_ir_headers = req
+            .state()
+            .schema.read().unwrap()
+            .sql_query(&cube_name, &ts_query);
 
-                if tc.len() != 2 {
-                    return boxed_error("Malformatted time cut".to_string());
-                }
+        let (query_ir, headers) = match query_ir_headers {
+            Ok(x) => x,
+            Err(err) => return boxed_error(err.to_string())
+        };
 
-                let time = match Time::from_key_value(tc[0].clone(), tc[1].clone()) {
-                    Ok(time) => time,
-                    Err(err) => return boxed_error(err.to_string())
-                };
+        info!("Query IR: {:?}", query_ir);
 
-                match cube_cache.clone() {
-                    Some(cache) => {
-                        let (cut, cut_value) = match cache.get_time_cut(time) {
-                            Ok(cut) => cut,
-                            Err(err) => return boxed_error(err.to_string())
-                        };
+        let sql = req.state()
+            .backend
+            .generate_sql(query_ir);
 
-                        agg_query.cuts = match agg_query.cuts {
-                            Some(mut cuts) => {
-                                cuts.insert(cut, cut_value);
-                                Some(cuts)
-                            },
-                            None => {
-                                let mut m: HashMap<String, String> = HashMap::new();
-                                m.insert(cut, cut_value);
-                                Some(m)
-                            },
-                        }
-                    },
-                    None => (),
-                };
+        info!("SQL query: {}", sql);
+
+        // Substitute header names (only need to do this once)
+        if final_headers.len() == 0 {
+            for header in &headers {
+                let mut new_header = header.clone();
+
+                for (k, v) in header_map.iter() {
+                    if header.contains(k) {
+                        new_header = new_header.replace(k, v);
+                    }
+                }
+
+                final_headers.push(new_header);
             }
-        },
-        None => (),
-    }
+        }
 
-    info!("Aggregate query: {:?}", agg_query);
+        sql_strings.push(sql);
+    }
 
-    // Turn AggregateQueryOpt into TsQuery
-    let ts_query: Result<TsQuery, _> = agg_query.try_into();
-    let ts_query = match ts_query {
-        Ok(q) => q,
-        Err(err) => return boxed_error(err.to_string())
-    };
+    info!("Headers: {:?}", final_headers);
+
+    // Joins all the futures for each TsQuery
+    let futures: JoinAll<Vec<Box<Future<Item=DataFrame, Error=Error>>>> = join_all(sql_strings
```

Comment:
small nitpick: because `futures` is also the name of the module, this might cause a naming conflict in the future. I'd prefer `futs`, but let me know what you think.